<?xml version="1.0"?>
<launch>
  <!-- Arguments -->
  <arg name="model"             default="raspicat"/>
  <arg name="cmd_vel_topic"     default="/cmd_vel"/>
  <arg name="odom_topic"        default="/odom"/>

  <arg name="waypoint_nav"      default="false"/>
  <arg name="move_forward_only" default="false"/>

  <!-- Move_base -->
  <node pkg="move_base" type="move_base" name="move_base" clear_params="true" respawn="true" output="screen">
    <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS"/>
    <rosparam command="load" ns="global_costmap" file="$(find raspicat_navigation)/config/param/costmap_common_params_$(arg model).yaml"/>
    <rosparam command="load" ns="global_costmap" file="$(find raspicat_navigation)/config/param/global_costmap_params.yaml"/>
    <rosparam command="load" ns="local_costmap" file="$(find raspicat_navigation)/config/param/costmap_common_params_$(arg model).yaml"/>
    <rosparam command="load" ns="local_costmap" file="$(find raspicat_navigation)/config/param/local_costmap_params.yaml"/>
    <rosparam command="load" file="$(find raspicat_navigation)/config/param/move_base_params.yaml"/>
    <rosparam command="load" file="$(find raspicat_navigation)/config/param/dwa_local_planner_params_$(arg model).yaml"/>
    <remap from="cmd_vel" to="$(arg cmd_vel_topic)"/>
    <remap from="odom"    to="$(arg odom_topic)"/>
    <param if="$(arg move_forward_only)" name="DWAPlannerROS/min_vel_x" value="0.0"/>
    
    <param if="$(arg waypoint_nav)" name="global_costmap/static_layer/map_topic"     value="/move_base_map"/>
    <param if="$(arg waypoint_nav)" name="local_costmap/static_layer/map_topic"      value="/move_base_map"/>
    <param if="$(arg waypoint_nav)" name="global_costmap/obstacles_layer/scan/topic" value="/filter_scan"/>
    <param if="$(arg waypoint_nav)" name="local_costmap/obstacles_layer/scan/topic"  value="/filter_scan"/>
  </node>
</launch>
